doc/user-guide.md

```diff
@@ -61,11 +61,32 @@ The Manager will automatically register the scheme for all custom resources defi
 
 The Manager can restrict the namespace that all controllers will watch for resources:
 ```Go
-mgr, err := manager.New(cfg, manager.Options{Namespace: namespace})
+// Create a new Cmd to provide shared dependencies and start components
+mgr, err := manager.New(cfg, manager.Options{
+   Namespace: namespace,
+   MapperProvider:     restmapper.NewDynamicRESTMapper,
+   MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+})
 ```
 By default this will be the namespace that the operator is running in. To watch all namespaces leave the namespace option empty:
 ```Go
-mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
+// Create a new Cmd to provide shared dependencies and start components
+mgr, err := manager.New(cfg, manager.Options{
+   Namespace: "",
+   MapperProvider:     restmapper.NewDynamicRESTMapper,
+   MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+})
+```
+Also, it can restrict a list of namespaces that all controllers will watch for resources:
```

Comment:
Hi @joelanford,

Thank you for raise this concern. If it can cause performance issues I think we should add it in the doc instead of not add this info at all. 

The reason for that is because in my understanding in a lot of scenarios users would like to use it because of security reasons.  I mean, it is a way to restrict the namespaces which the operator should watch/do actions.

I will try to address your comment/concern.

Reply:
I'm on board with adding something to the docs about the potential problems with multi-namespace watches. Since that's a feature of controller-runtime, I think it would be better suited upstream, probably in the controller-runtime godoc and perhaps also in the kubebuilder book.

Since we're trying to consolidate documentation, I think we should avoid creating docs about controller-runtime features in the SDK repo, and instead make sure we have prominent links to upstream docs.

Reply:
I understand your point. The problem in POV about we do not add the info here is:
- It is shows one of the 10 questions
- we can address comments here https://godoc.org/sigs.k8s.io/controller-runtime/pkg/cache#NewCacheFunc, however, it is 3 levels down which will be very hard for the users and us indeed to know that it exists and to get the info. 

So, IMHO:  just address there will still making this subject came back in the issues/ml :-(

Reply:
> It shows one of the 10 questions

Are you referring to this: https://github.com/operator-framework/operator-sdk/issues/767#issuecomment-530980302? If so, please link to that issue in your description for #2008 and #2010.   

> very hard for the users and us indeed to know that exists

This is kind of what I'm getting at 🙂. My thought is that we probably shouldn't make multi-namespace support obvious to novice users in our user guide because they're not as likely to understand that it can be used incorrectly, even if we include the disclaimers.

More broadly, this section about the manager is really controller-runtime documentation, not SDK documentation. We should not maintain separate documentation from controller-runtime about controller-runtime APIs because it is extremely likely to get out of date.

Like I said, I'm okay with documenting the multi-namespace cache, but I think it should be in upstream docs, which we can link to. 

I think if we update the GoDoc [here](https://github.com/kubernetes-sigs/controller-runtime/blob/59b131b7cd54d56ec74a66b084a53dd1c3e4843f/pkg/cache/multi_namespace_cache.go#L37-L39) and [here](https://github.com/kubernetes-sigs/controller-runtime/blob/59b131b7cd54d56ec74a66b084a53dd1c3e4843f/pkg/cache/multi_namespace_cache.go#L59-L62) to improve the description, add the disclaimers, and include an example that would be a great start. At that point, maybe we could add a line about it in the `Advanced Topics` section.

@shawn-hurley what are your thoughts on this? Was there/is there discussion about backing multi-namespace watches out of controller-runtime?

Reply:
Hi @joelanford,

I spoke with @shawn-hurley and it very safe to be used. According to him `It will only be an issue if your api-server is super flooded or you are watching 10000 namespaces` 

I update the note as well.